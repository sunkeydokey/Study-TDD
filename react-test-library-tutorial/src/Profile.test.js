import React from 'react';
import { render, screen } from '@testing-library/react';
import Profile from './Profile';

describe('Profile', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="sunkeydokey" name="sunkee" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the texts correctly', () => {
    render(<Profile username="sunkeydokey" name="sunkee" />);
    screen.getByText('sunkeydokey');
    screen.getByText('(sunkee)');
    screen.getByText(/sunkee/);
  });
});
