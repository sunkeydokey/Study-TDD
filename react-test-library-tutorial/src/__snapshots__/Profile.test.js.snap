// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches snapshot 1`] = `
<div>
  <div>
    <b>
      sunkeydokey
    </b>
    Â 
    <span>
      (
      sunkee
      )
    </span>
  </div>
</div>
`;
